Source/WebCore/platform/graphics/egl/GLContext.cpp

```diff
@@ -260,21 +260,24 @@ std::unique_ptr<GLContext> GLContext::createWindowContext(GLNativeWindowType win
         surface = createWindowSurfaceX11(display, config, window);
         break;
 #endif
-#if PLATFORM(WAYLAND)
-    case PlatformDisplay::Type::Wayland:
-        surface = createWindowSurfaceWayland(display, config, window);
-        break;
-#endif
 #if USE(WPE_RENDERER)
     case PlatformDisplay::Type::WPE:
         surface = createWindowSurfaceWPE(display, config, window);
         break;
 #endif // USE(WPE_RENDERER)
+#if PLATFORM(WAYLAND)
+    case PlatformDisplay::Type::Wayland:
+        RELEASE_ASSERT_NOT_REACHED();
+        break;
+#endif
 #if USE(GBM)
     case PlatformDisplay::Type::GBM:
+        RELEASE_ASSERT_NOT_REACHED();
+        break;
 #endif
     case PlatformDisplay::Type::Surfaceless:
         RELEASE_ASSERT_NOT_REACHED();
+        break;
     }
```

Comment:
How about:

```suggestion
    default:
        RELEASE_ASSERT_NOT_REACHED();
        break;
    }
```

Or alternatively, if you want a compiler warning whenever a new PlatformDisplay::Type is added:

```suggestion
    }

    RELEASE_ASSERT(surface);
```

Reply:
Yes, I don't wat to use default: and I don't understand the second suggestion

Reply:
I would rather do this:

```suggestion
#if PLATFORM(WAYLAND)
    case PlatformDisplay::Type::Wayland:
#endif
#if USE(GBM)
    case PlatformDisplay::Type::GBM:
#endif
    case PlatformDisplay::Type::Surfaceless:
        RELEASE_ASSERT_NOT_REACHED();
        break;
    }
```

This is borth avoiding the duplicated code, and without using a `default:` case.

Reply:
That's what I had initially, but I wasn't sure it was clear enough with all the ifdefs